hummingbot/connector/gateway_EVM_AMM.py

```diff
@@ -155,6 +156,13 @@ def is_pending_approval(self, token: str) -> bool:
         pending_approval_tokens = [tk.split("_")[2] for tk in self._in_flight_orders.keys()]
         return True if token in pending_approval_tokens else False
 
+    def _update_connector_fees(self, token: str, gas_price: Decimal, is_maker: bool):
+        connector_setting = AllConnectorSettings[self.name]
+        if is_maker:
+            connector_setting.trade_fee_schema.maker_fixed_fees = [TokenAmount(token, gas_price)]
+        else:
+            connector_setting.trade_fee_schema.taker_fixed_fees = [TokenAmount(token, gas_price)]
```

Comment:
The fee you need to pay is not just `gas_price` - it's `gas_price` * `gas_cost`.

Reply:
Let's use a real swap transaction as an example

https://ropsten.etherscan.io/tx/0xeaf6b51033b8060c527d58f71ec66226da8d98ec77290017ec7787ed7ab60ae6

The code that emitted the transaction specified a gas limit of 150688, which is not very precise but also not too far from the actual gas amount used (98662).

The gas price for each unit of gas was 30 Gwei. 1 wei is 1e-18 ETH.

So the actual transaction fee spent was `30e9 * 98662 * 1e18 = 0.00295986 ETH`.

Now, the actual gas use is only really known when the transaction is executed. Let's say you're just estimating and you're using the gas limit, or some bigger number that you got from the Uniswap libary. Then, you can estimate the transaction cost like this.

`${gas price in wei} * ${gas estimate} * 1e-18`

So if you're using the gas limit instead, then the cost estimate would be

`150688 * 30e9 * 1e-18 = 0.00452064 ETH`.

Reply:
This also highlights a flaw with this design... there's actually no fixed fee for a `GatewayEVMAMM` connector - the gas limits for, say, Uniswap V2 and Uniswap V3 are different (with Uniswap v3 being much higher). So the fees for trading on Uniswap v3 are higher, even with the same connector.

Reply:
Let's just make the gas fees calculation a special case for `GatewayEVMAMM` for now - without needing to following prior patterns for trading fees. The way it works for gateway exchanges is different.